### PR TITLE
code.angularjs.org / Docs App updates

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
load the example files in the docs app based on the current deployment

I'm getting these warnings in dgeni with my changes: 

```
warn:    No getAlias(doc) method provided - doc "deployment-data-debug" (deployment-data)
warn:    No getAlias(doc) method provided - doc "deployment-data-default" (deployment-data)
warn:    No getAlias(doc) method provided - doc "deployment-data-jquery" (deployment-data)
warn:    No getAlias(doc) method provided - doc "deployment-data-production" (deployment-data)

warn:    No path template provided - doc "deployment-data-debug" (deployment-data)
warn:    No path template provided - doc "deployment-data-default" (deployment-data)
warn:    No path template provided - doc "deployment-data-jquery" (deployment-data)
warn:    No path template provided - doc "deployment-data-production" (deployment-data)
```

### CODE REVIEW

File: scripts/code.angularjs.org-firebase/functions/index.js [hunk 1]
```diff
@@ -32,43 +30,148 @@ function sendStoredFile(request, response) {
...
+    if (path !== '/') getFilesOptions.prefix = path;
+
+    let fileList = [];
+    let directoryList = [];
+
+    return getContent(getFilesOptions).then(() => {
+      let contentList = '';
+
+      directoryList.forEach(directoryPath => {
+        const dirName = directoryPath.split('/').reverse()[1];
```
Comment (gkalpak) [inline scripts/code.angularjs.org-firebase/functions/index.js]: :smiley:

File: scripts/code.angularjs.org-firebase/functions/index.js [hunk 2]
```diff
@@ -32,43 +30,148 @@ function sendStoredFile(request, response) {
...
+        contentList += `<a href="${dirName}/">${dirName}/</a><br>`;
+      });
+
+      fileList.forEach(file => {
+        const fileName = file.metadata.name.split('/').reverse()[0];
```
Comment (gkalpak) [inline scripts/code.angularjs.org-firebase/functions/index.js]: Or: `file.metadata.name.split('/').pop()`

File: scripts/code.angularjs.org-firebase/functions/index.js [hunk 3]
```diff
@@ -32,43 +30,148 @@ function sendStoredFile(request, response) {
...
+        .send(directoryListing);
     });
+
+    function getContent(options) {
+      return bucket.getFiles(options).then(data => {
+        const files = data[0];
+        const nextQuery = data[1];
+        const apiResponse = data[2];
+
+        if (files.length === 0 && (apiResponse && !apiResponse.prefixes || !apiResponse)) {
```
Comment (gkalpak) [inline scripts/code.angularjs.org-firebase/functions/index.js]: Or: `if (!files.length && (!apiResponse || !apiResponse.prefixes)`

**Review by gkalpak — CHANGES_REQUESTED** [review]
Unfortunately, there is no way to actually test this, but it mostly LGTM (with Travis issues fixed).

File: docs/config/templates/examples/runnableExample.template.html [hunk 4]
```diff
@@ -18,7 +18,7 @@
     </div>
   {% endfor %}
 
-    <iframe class="runnable-example-frame" src="{$ doc.example.deployments.default.outputPath $}" name="{$ doc.example.id $}"></iframe>
+    <iframe class="runnable-example-frame" ng-src="{{getExampleIndex('{$ doc.example.deployments.default.path $}', deployment)}}" name="{$ doc.example.id $}"></iframe>
```
Comment (Narretz) [inline docs/config/templates/examples/runnableExample.template.html]: Can't tell if you're joking ;) ngSrc takes interpolation :/

Comment (gkalpak) [inline docs/config/templates/examples/runnableExample.template.html]: OMG! Confused by Angular's `[src]`. That's embarassing 😳
The second comment is still valid (hopefully) :grin:

**Comment by Narretz** [comment]
Second one is fixed ;)

George Kalpakas <notifications@github.com> schrieb am Mi., 26. Juli 2017
20:13:

**Comment by Narretz** [comment]
Is this okay now @gkalpak ?

**Review by gkalpak — APPROVED** [review]

**Comment by gkalpak** [comment]
I think so :smiley:

**Pull request merged**